Backend/webblog/blog/views.py

```diff
@@ -1,29 +1,50 @@
 from django.contrib.auth.models import User
+from django.utils import timezone
 from rest_framework import viewsets, permissions
+from rest_framework.response import Response
 
-from blog.models import Blog
-from blog.serializers import BlogSerializer, UserSerializer
+from blog.models import Blog, Comment
+from blog.permissions import IsOwnerOrReadOnly
+from blog.serializers import BlogSerializer, UserSerializer, CommentSerializer
 
 
 class BlogAPI(viewsets.ModelViewSet):
     queryset = Blog.objects.all()
 
-    permission_classes = [
-        permissions.IsAuthenticated
-    ]
+    permission_classes = [permissions.IsAuthenticatedOrReadOnly,
+                          IsOwnerOrReadOnly]
 
     serializer_class = BlogSerializer
 
-    def get_queryset(self):
-        return self.request.user.blog.all()
-
     def perform_create(self, serializer):
         serializer.save(owner=self.request.user)
 
+    def perform_update(self, serializer):
+        serializer.save(created=timezone.now())
+
+    def list(self, request, *args, **kwargs):
+        if request.GET.get('title', '') != '':
+            queryset = Blog.objects.filter(title__icontains=request.GET.get('title'))
```

Comment:
From fronend we should expect only one argument `search_keyword` and on backend here we should find all the blogs which have those keywords in title or in the description.

Hint: Use Q objects for query: https://docs.djangoproject.com/en/2.2/topics/db/queries/#complex-lookups-with-q-objects

Reply:
Done.
I've committed the changes.